### PR TITLE
Add Documentation

### PR DESCRIPTION
**Pull request by Sachindu-Nethmin**
Add Documentation

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [x] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,17 +14,91 @@ The `ballerinax/spotify` package provides APIs to connect and interact with Spot
...
+
+```bash
+    curl -X POST "https://accounts.spotify.com/api/token" \
+     -H "Content-Type: application/x-www-form-urlencoded" \
+     -d "grant_type=client_credentials&client_id=your-client-id&client_secret=your-client-secret"
+```
+The response will return an access token valid for 1 hour
+```
+{
+  "access_token": "BQDBKJ5eo5jxbtpWjVOj7ryS84khybFpP_lTqzV7uV-T_m0cTfwvdn5BnBSKPxKgEb11",
```
Comment (aashikam) [inline README.md:51]: Just add 
```suggestion
  "access_token": "<Access Token>",
```
And this should be enough. I saw a screenshot also with some client details, lets blur/hide them.

**Review by aashikam — COMMENTED** [review]
@Sachindu-Nethmin Hi, I added a small comment, rest of it looks fine.

**Comment by Sachindu-Nethmin** [comment]
@aashikam please review

**Comment by aashikam** [comment]
@Sachindu-Nethmin Hey

Hiding sensitive information such as access tokens and client IDs in Git pull requests is critical to prevent unauthorized access and potential security breaches. Exposing such credentials can allow attackers to exploit your application, access restricted resources, or compromise user data, leading to serious damage.

7de1d1a1f1bdc853e71537ddeda32cb6f00551d2  This commit itself is a big security vulnerability. 😃

Even though you have removed it, by viewing the commit anyone still has access to the information. 

To remove this from git history you will have to do a hard reset. Here are the steps:

1. Check the commit history: Run `git log` to view your commit history and identify the commit hash (e.g., abc123) you want to reset to.
2. Perform the hard reset: Execute `git reset --hard <commit-hash>` to move the branch pointer to the specified commit and discard any changes in the working directory or staging area.
3. Verify the reset: Run `git log` again to confirm the branch now points to the desired commit.
4. Push the changes: Use `git push --force` to overwrite the remote branch

After that please add an example. Here is a sample PR: https://github.com/ballerina-platform/module-ballerinax-dropbox/pull/6

**Comment by aashikam** [comment]
@Sachindu-Nethmin Hey

Documentation includes all these files: 
Screenshot 2024-12-06 at 09 58 44

I see that you have only updated the README.md. It's pretty much the same content. 
Here is a sample PR: https://github.com/ballerina-platform/module-ballerinax-dropbox/pull/5

**Pull request closed without merging**